crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -292,7 +299,10 @@ impl RuffSettingsIndex {
                             }
                         }
                     }
-                    Ok(None) => {}
+                    Ok(None) => {
+                        let settings = RuffSettings::editor_only(editor_settings, &directory);
+                        index.write().unwrap().insert(directory, Arc::new(settings));
+                    }
```

Comment:
If I'm understanding this change correctly, we'd now have a fallback settings for _every_ directory in the project. I think this might create an unexpected behavior because for a file in a nested directory, it would use this fallback settings instead of the settings from the project root? Let me test this out.

Reply:
Yeah, I tested out, I think this is an incorrect fix. Consider the following directory structure:
```
.
├── nested
│   └── foo
│       └── test.py
├── pyproject.toml
└── test.py
```

The `nested/foo/test.py` does not consider any settings from `pyproject.toml`, it's only the `test.py` file that will consider it.